ArduPlane/mode_autoland.cpp

```diff
@@ -175,7 +175,11 @@ bool ModeAutoLand::_enter()
     entry_alt = plane.current_loc.alt;
     
     // start first leg toward the base leg loiter to alt point
-    if (climb_first) {
+    if (climb_first && plane.control_mode_reason != ModeReason::BATTERY_FAILSAFE && !(plane.fence.get_breaches()
+#if AP_FENCE_ENABLED
+ & AC_FENCE_TYPE_ALT_MAX)
+#endif
```

Comment:
I disagree with this, better to keep behaviour consistent. Mode should behave the same no matter how you enter them.

Reply:
I agree with @IamPete1, it should be consistent

Reply:
fixed